crates/isograph_schema/src/validate_schema.rs

```diff
@@ -968,13 +977,63 @@ pub fn categorize_field_loadability<'a>(
     }
 }
 
-pub fn get_missing_arguments_and_validate_argument_types<'a>(
+fn validate_no_extraneous_arguments(
+    argument_definitions: &[&ValidatedVariableDefinition],
+    arguments: &[WithLocation<SelectionFieldArgument>],
+    location: Location,
+) -> ValidateSelectionsResult<()> {
+    let extra_arguments: Vec<_> = arguments
+        .iter()
+        .filter_map(|arg| {
+            // TODO remove this
+            // With @exposeField on Query, id field is needed because the generated
+            // query is like node(id: $id) { ... everything else }, but that
+            // id field is added in somewhere else
+            if arg.item.name.item.lookup() == "id" {
```

Comment:
nit: let's define a variable up top `let foo = "id".intern().into()` and compare `arg.item.name.item == foo` so that we do work on every iteration. We could also move it into a `lazy_static!` block, which would allow us to reuse the same value across calls to `validate_no_extraneous_arguments`

Reply:
e.g. https://github.com/isographlabs/isograph/blob/d8aff7180684e69152eb93578b59673be4e54159/crates/graphql_artifact_generation/src/generate_artifacts.rs#L44